.changeset/rare-pillows-fold.md

```diff
@@ -0,0 +1,5 @@
+---
+'xstate': patch
+---
+
+The initial state is now cached inside of the service instance instead of the machine, which was the previous (faulty) strategy. This will prevent entry actions on initial states from being called more than once, which is important for ensuring that actors are not spawned more than once.
```

Comment:
The added test doesn't quite reflect this part:
> which is important for ensuring that actors are not spawned more than once.

Could we add a test (or edit the one already added) for exactly what is described here?

Reply:
Executing the assignment function in the provided test is the only way an actor can potentially be spawned multiple times. It is sufficient to test that the assignment function is only called once.

Reply:
Yes - it’s sufficient but the test case doesnt make this particular intention clear. Even putting the aside the implementation of this test - it’s title only mentions caching. Im also not immediately sure how reading initialState lead to executing its actions (spawning in here) as it is supposed to be pure so I’m not sure how caching has helped here as it shouldnt matter much.